src/Item/MenuItem.php

```diff
@@ -212,6 +213,37 @@ public function getParent () : ?MenuItem
     }
 
 
+    /**
+     * @param MenuItem|null $parent
+     *
+     * @return MenuItem
+     */
+    public function setParent (?MenuItem $parent) : self
+    {
+        // remove item from parent's children
+        if (null !== $this->parent)
+        {
```

Comment:
I'd prefer an `MenuItem::removeChildren(Menuitem $item)` method that we call here.

Reply:
We don't want to have this public method.

Reply:
I just added it, and also expanded a lot of the different tree manipulation methods, so that they are now fully-featured and (hopefully) stable.

Reply:
Awesome! 👍